pandas/io/tests/test_excel.py

```diff
@@ -1328,6 +1328,20 @@ def test_to_excel_multiindex(self):
                             parse_dates=False)
             tm.assert_frame_equal(frame, df)
 
+    # GH13511
+    def test_to_excel_multiindex_nan_label(self):
+        _skip_if_no_xlrd()
+
+        frame = self.frame
+        frame.A = np.arange(len(frame))
```

Comment:
use ``range(len(frame))`` again this will have a comparison issue on other platforms (e.g. windows). The reason is numpy defaults to int32's, while pandas uses int64's. So comparison are off if you are directly constructing.